app/health/checks.go

```diff
@@ -0,0 +1,100 @@
+// Copyright © 2022-2023 Obol Labs Inc. Licensed under the terms of a Business Source License 1.1
+
+package health
+
+type severity string
+
+const (
+	severityCritical severity = "critical"
+	severityWarning  severity = "warning"
+	severityInfo     severity = "info"
+)
+
+// checkType indicate whether the check is a positive or a negative check.
+// A positive check is considered successful if the metric is 1.
+// A negative check is considered successful if the metric is 0.
+type checkType string
+
+const (
+	typePositive checkType = "positive"
+	typeNegative checkType = "negative"
+)
+
+type metadata struct {
+	NumValidators int
+	NumPeers      int
+	QuorumPeers   int
+}
+
+// check is a health check.
+type check struct {
+	Name        string
+	Description string
+	Type        checkType
+	Severity    severity
+	Func        func(store, metadata) bool
+}
+
+var checks = []check{
+	{
+		Name:        "error_logs",
+		Description: "Error logs detected that require human intervention.",
+		Type:        typeNegative,
+		Severity:    severityCritical,
+		Func: func(s store, _ metadata) bool {
+			counts := s.Get("app_log_error_total", noLabels)
+			return counterIncrease(counts) > 0
+		},
+	},
+	{
+		Name:        "high_warning_log_rate",
+		Description: "High rate of warning logs. Please check the logs for more details.",
+		Type:        typeNegative,
+		Severity:    severityCritical,
+		Func: func(s store, m metadata) bool {
+			counts := s.Get("app_log_error_total", noLabels)
+			return counterIncrease(counts) > 2*float64(m.NumValidators) // Allow 2 warnings per validator.
+		},
+	},
+	{
+		Name:        "beacon_node_syncing",
+		Description: "Beacon Node in syncing state.",
+		Type:        typeNegative,
+		Severity:    severityCritical,
+		Func: func(s store, m metadata) bool {
+			gaugeVals := s.Get("app_monitoring_beacon_node_syncing", noLabels)
+			return gaugeMax(gaugeVals) == 1
+		},
+	},
+	{
+		Name:        "insufficient_connected_peers",
+		Description: "Not connected to at least quorum peers. Check logs for networking issue or coordinate with peers.",
+		Type:        typePositive,
+		Severity:    severityCritical,
+		Func: func(s store, m metadata) bool {
+			counts := s.Get("p2p_peer_connection_total", countNonZeroLabels)
+			return gaugeMin(counts) < float64(m.QuorumPeers)
+		},
+	},
+	{
+		Name:        "pending_validators",
+		Description: "Pending validators detected. Activate them to start validating.",
+		Type:        typeNegative,
```

Comment:
would we still call this negative if there are no pending validators, ie, `gaugeMax(counts) == 0` ?